src/Aspire.Hosting/Postgres/PostgresBuilderExtensions.cs

```diff
@@ -62,6 +63,39 @@ public static IResourceBuilder<PostgresDatabaseResource> AddDatabase(this IResou
                                          .WithManifestPublishingCallback(context => WritePostgresDatabaseToManifest(context, postgresDatabase));
     }
 
+    /// <summary>
+    /// Adds a pgAdmin 4 administration and development platform for PostgreSQL to the application model.
+    /// </summary>
+    /// <param name="builder">The PostgreSQL server resource builder.</param>
+    /// <param name="hostPort">The host port for the application ui.</param>
+    /// <param name="containerName">The name of the container. Defaults to "pgAdmin".</param>
+    /// <param name="options"><see cref="PgAdminOptions"/> for the container (optional).</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{PostgresContainerResource}"/>.</returns>
+    public static IResourceBuilder<PostgresContainerResource> WithPgAdmin(this IResourceBuilder<PostgresContainerResource> builder, int? hostPort, string containerName = "pgAdmin", PgAdminOptions? options = null)
+    {
+        if (builder.ApplicationBuilder.Resources.OfType<PgAdminContainerResource>().Any())
+        {
+            return builder;
+        }
+
+        ArgumentNullException.ThrowIfNull(hostPort);
+
+        options = options ?? new();
+
+        var pgAdminContainer = new PgAdminContainerResource(containerName);
+        pgAdminContainer.Annotations.Add(new ContainerImageAnnotation { Image = "dpage/pgadmin4", Tag = "latest" });
+        pgAdminContainer.Annotations.Add(new ServiceBindingAnnotation(ProtocolType.Tcp, port: hostPort, containerPort: 80, uriScheme: "http", name: containerName));
+        pgAdminContainer.Annotations.Add(new EnvironmentCallbackAnnotation(context =>
```

Comment:
Is it possible to pre-configure the connection to the postgres database so devs can just click on the URL in the dashboard and everything is already connected.

Reply:
That would be ideal.

Reply:
Also add `.WithManifestPublishingCallback(ManifestPublishingCallbackAnnotation.Ignore)` as we probably don't want these containers appearing in the manifest.

Reply:
ExcludeFromManifest

Reply:
Some points to discuss about the logic of pre-configuring pgAdmin with the servers:

At the point where the implementation is done, I have no way of getting the host port associated with postgres so I had to rely on the ServiceBindingAnnotation. Consequently, if the developer does not specify a port, I have to ignore it and not generate the pre-configuration.

I believe we found the motivation to return to the previous approach of using LifecycleHook where I would have the port information already assigned and we would also gain the possibility of pre-configuring all possible Postgres servers.

Regarding how to generate the json file that needs to be mounted in the container, I am generating a temporary file for this purpose and I am not sure if this is the best option.

Finally we have the issue of having to use "host.docker.internal" as the postgres server host. I couldn't find another way and I believe it doesn't work on Linux.

Reply:
Looks like this lifecycle hook is in now.

Reply:
I did an experiment with the @davidfowl new lifecycle hook but at the point where `AfterEndpointsAllocatedAsync` is called if I add a ContainerResource to the appModel it is not executed anymore.

This would be the moment when I can get all the postgres servers to pre-configure pgadmin4 with connections to them, but I cannot initialize the pgadmin4 container at this point.

Therefore, the implementation remains as it is, and it is only possible to pre-configure pgadmin4 with the postgres server if the developer explicitly specifies a port.

What direction do you want me to take?